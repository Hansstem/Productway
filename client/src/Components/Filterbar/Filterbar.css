.filterChips {
	display: flex;
	gap: 10px;
	margin-left: 40px;
	flex-direction: column;
	margin-top: 20px;
	margin-bottom: 20px;
	max-width: 90%;
}

.filterChips h3 {
	margin: 0px;
}

.chips {
	margin-top: 0px;
	display: flex;
	flex-wrap: wrap;
	gap: 10px;
	align-items: center;
}

.filter {
	display: flex;
	gap: 10px;
	margin-right: 30px;
}

.filtercontainer {
	margin-top: 0px;
	display: flex;
	flex-wrap: wrap;
	gap: 10px;
	align-items: center;
}

.chips .MuiChip-root {
	transition:
		transform 0.1s,
		box-shadow 0.1s,
		background-color 0.1s;
	box-shadow: 0 0 5px rgba(100, 100, 100, 0.1);
}

.chips .MuiChip-root:hover {
	transform: scale(1.1);
	box-shadow: 0 0 10px rgba(100, 100, 100, 0.4);
}

.filter .MuiChip-root {
	transition:
		transform 0.1s,
		box-shadow 0.1s,
		background-color 0.1s;
	box-shadow: 0 0 5px rgba(100, 100, 100, 0.2);
}

.filter .MuiChip-root:hover {
	transform: scale(1.1);
	box-shadow: 0 0 10px rgba(100, 100, 100, 0.4);
}

.chips .MuiChip-clickable.MuiChip-clickable:active {
	background-color: #1c326e49;
}

.sliderContainer {
	display: flex;
	align-items: center;
	gap: 20px;
	text-align: center;
	max-width: 90%;
}

.sliderContainer p {
	margin-top: 0px;
	margin-bottom: 0px;
	font-size: 15px;
}

.slider {
	margin-top: 6px;
	width: 250px;
}

.slider .MuiSlider-track {
	background-color: #1c326e49;
	border-color: #1c326e49;
}

.searchTerm {
	display: flex;
	gap: 10px;
}

.searchTerm Button {
	margin-top: auto;
	margin-bottom: auto;
}
