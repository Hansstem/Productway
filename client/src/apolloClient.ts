import { ApolloClient, InMemoryCache } from '@apollo/client';

export const client = new ApolloClient({
	uri: 'http://it2810-10.idi.ntnu.no:4000/graphql',
	credentials: 'include',
	cache: new InMemoryCache(),
});
