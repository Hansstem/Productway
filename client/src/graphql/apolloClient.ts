import { ApolloClient, InMemoryCache } from '@apollo/client';

export const client = new ApolloClient({
	uri: 'http://it2810-10.idi.ntnu.no:4000/graphql', // for production
	//uri: 'http://localhost:4000/graphql', // for local development
	credentials: 'include',
	cache: new InMemoryCache(),
});
