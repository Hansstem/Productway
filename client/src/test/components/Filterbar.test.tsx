import { StrictMode } from 'react';

import { ApolloProvider } from '@apollo/client';
import {
	fireEvent,
	render,
	RenderResult,
	screen,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { RecoilRoot } from 'recoil';

import Filterbar from '../../components/Filterbar/Filterbar';
import { client } from '../../graphql/apolloClient';

describe('Header test', () => {
	let filterbar: RenderResult;
	beforeEach(() => {
		filterbar = render(
			<StrictMode>
				<RecoilRoot>
					<ApolloProvider client={client}>
						<MemoryRouter>
							<Filterbar />
						</MemoryRouter>
					</ApolloProvider>
				</RecoilRoot>
			</StrictMode>,
		);
	});

	it('snapshottest filterbar', () => {
		expect(filterbar).toMatchSnapshot();
	});

	it('renders correctly', () => {
		expect(filterbar.getAllByText(/Kategori/i)).toBeTruthy();
		expect(filterbar.getAllByText(/Sortering/i)).toBeTruthy();
		expect(filterbar.getAllByText(/Minimal pris/i)).toBeTruthy();
		expect(filterbar.getAllByText(/Maksimal pris/i)).toBeTruthy();

		expect(filterbar.getAllByText(/Snacks & Godteri/i)).toBeTruthy();
		expect(filterbar.getAllByText(/Personlige artikler/i)).toBeTruthy();
		expect(filterbar.getAllByText(/Kjøtt/i)).toBeTruthy();
		expect(filterbar.getAllByText(/Middag/i)).toBeTruthy();
		expect(filterbar.getAllByText(/Ost/i)).toBeTruthy();
		expect(filterbar.getAllByText(/Dessert/i)).toBeTruthy();
		expect(filterbar.getAllByText(/Pålegg & frokost/i)).toBeTruthy();
		expect(filterbar.getAllByText(/Middagstilbehør/i)).toBeTruthy();
		expect(filterbar.getAllByText(/Drikke/i)).toBeTruthy();

		expect(filterbar.getAllByText(/Stigende pris/i)).toBeTruthy();
		expect(filterbar.getAllByText(/Synkende pris/i)).toBeTruthy();
	});

	it('changes color when clicked', () => {
		const snacksAndCandyButton = screen.getByRole('button', {
			name: 'Snacks & godteri',
		});
		expect(snacksAndCandyButton).toHaveStyle(
			'background-color: rgba(0, 0, 0, 0.04); color: rgba(0, 0, 0, 0.87);',
		);
		fireEvent.click(snacksAndCandyButton);
		expect(snacksAndCandyButton).toHaveStyle(
			'background-color: rgb(40, 112, 148); color: rgb(255, 255, 255);',
		);
		fireEvent.click(snacksAndCandyButton);
		expect(snacksAndCandyButton).toHaveStyle(
			'background-color: rgba(0, 0, 0, 0.04); color: rgba(0, 0, 0, 0.87);',
		);
		fireEvent.click(snacksAndCandyButton);
	});

	it('toggles the showStatus property of a category chip when clicked', () => {
		const snacksAndCandyChip = screen.getByRole('button', {
			name: 'Snacks & godteri',
		});
		expect(snacksAndCandyChip).toHaveClass(
			'MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-8m5rrr-MuiButtonBase-root-MuiChip-root',
		);
		fireEvent.click(snacksAndCandyChip);
		expect(snacksAndCandyChip).toHaveClass(
			'MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root',
		);
		fireEvent.click(snacksAndCandyChip);
		expect(snacksAndCandyChip).toHaveClass(
			'MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-8m5rrr-MuiButtonBase-root-MuiChip-root',
		);
	});

	it('toggles the showStatus property of a sorting filter chip when clicked', () => {
		const ascendingPriceChip = screen.getByRole('button', {
			name: 'Stigende pris',
		});
		expect(ascendingPriceChip).toHaveClass('MuiChip-outlined');
		fireEvent.click(ascendingPriceChip);
		expect(ascendingPriceChip).toHaveClass('MuiChip-filled');
	});

	it('untoggles the showStatus property of a sorting filter when the other sorting filter is clicked', () => {
		const ascendingPriceChip = screen.getByRole('button', {
			name: 'Stigende pris',
		});
		const descendingPriceChip = screen.getByRole('button', {
			name: 'Synkende pris',
		});
		expect(ascendingPriceChip).toHaveClass(
			'MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-8m5rrr-MuiButtonBase-root-MuiChip-root',
		);
		expect(descendingPriceChip).toHaveClass('MuiChip-outlined');
		fireEvent.click(ascendingPriceChip);
		expect(ascendingPriceChip).toHaveClass(
			'MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root',
		);
		expect(descendingPriceChip).toHaveClass('MuiChip-outlined');
		fireEvent.click(descendingPriceChip);
		expect(ascendingPriceChip).toHaveClass('MuiChip-outlined');
		expect(descendingPriceChip).toHaveClass('MuiChip-filled');
	});
});
