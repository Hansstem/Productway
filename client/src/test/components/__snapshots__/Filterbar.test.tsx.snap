// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header test > snapshottest filterbar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filterChips"
      >
        <h3>
          Kategori
        </h3>
        <div
          class="chips"
        >
          <div
            aria-label="Snacks & godteri"
            class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
            data-testid="Snacks & godteri"
            role="button"
            tabindex="0"
          >
            <i
              class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-popcorn"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Snacks & godteri
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            aria-label="Personlige artikler"
            class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
            data-testid="Personlige artikler"
            role="button"
            tabindex="0"
          >
            <i
              class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-soap"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Personlige artikler
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            aria-label="Kjøtt"
            class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
            data-testid="Kjøtt"
            role="button"
            tabindex="0"
          >
            <i
              class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-meat"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Kjøtt
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            aria-label="Middag"
            class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
            data-testid="Middag"
            role="button"
            tabindex="0"
          >
            <i
              class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-restaurant"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Middag
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            aria-label="Ost"
            class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
            data-testid="Ost"
            role="button"
            tabindex="0"
          >
            <i
              class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-cheese"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Ost
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            aria-label="Dessert"
            class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
            data-testid="Dessert"
            role="button"
            tabindex="0"
          >
            <i
              class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rs-pie"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Dessert
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            aria-label="Pålegg & frokost"
            class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
            data-testid="Pålegg & frokost"
            role="button"
            tabindex="0"
          >
            <i
              class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-bread-slice"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Pålegg & frokost
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            aria-label="Middagstilbehør"
            class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
            data-testid="Middagstilbehør"
            role="button"
            tabindex="0"
          >
            <i
              class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-sauce"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Middagstilbehør
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            aria-label="Drikke"
            class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
            data-testid="Drikke"
            role="button"
            tabindex="0"
          >
            <i
              class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-drink-alt"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Drikke
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </div>
        <h3>
          Sortering
        </h3>
        <div
          class="filtercontainer"
        >
          <div
            class="filter"
          >
            <div
              aria-label="Stigende pris"
              class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
              data-testid="Stigende pris"
              role="button"
              tabindex="0"
            >
              <i
                class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-arrow-trend-up"
              />
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                Stigende pris
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
            <div
              aria-label="Synkende pris"
              class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
              data-testid="Synkende pris"
              role="button"
              tabindex="0"
            >
              <i
                class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-arrow-trend-down"
              />
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                Synkende pris
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </div>
          <div
            class="sliderContainer"
          >
            <p>
               Minimal pris
            </p>
            <div
              class="slider"
            >
              <span
                class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-187mznn-MuiSlider-root"
              >
                <span
                  class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
                />
                <span
                  class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
                  style="left: 0%; width: 100%;"
                />
                <span
                  class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
                  data-index="0"
                  style="left: 0%;"
                >
                  <input
                    aria-orientation="horizontal"
                    aria-valuemax="200"
                    aria-valuemin="0"
                    aria-valuenow="0"
                    aria-valuetext="0"
                    data-index="0"
                    max="200"
                    min="0"
                    step="1"
                    style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                    type="range"
                    value="0"
                  />
                  <span
                    aria-hidden="true"
                    class="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
                  >
                    <span
                      class="MuiSlider-valueLabelCircle"
                    >
                      <span
                        class="MuiSlider-valueLabelLabel"
                      >
                        0
                      </span>
                    </span>
                  </span>
                </span>
                <span
                  class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
                  data-index="1"
                  style="left: 100%;"
                >
                  <input
                    aria-orientation="horizontal"
                    aria-valuemax="200"
                    aria-valuemin="0"
                    aria-valuenow="200"
                    aria-valuetext="200"
                    data-index="1"
                    max="200"
                    min="0"
                    step="1"
                    style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                    type="range"
                    value="200"
                  />
                  <span
                    aria-hidden="true"
                    class="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
                  >
                    <span
                      class="MuiSlider-valueLabelCircle"
                    >
                      <span
                        class="MuiSlider-valueLabelLabel"
                      >
                        200
                      </span>
                    </span>
                  </span>
                </span>
              </span>
            </div>
            <p>
              Maksimal pris
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filterChips"
    >
      <h3>
        Kategori
      </h3>
      <div
        class="chips"
      >
        <div
          aria-label="Snacks & godteri"
          class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
          data-testid="Snacks & godteri"
          role="button"
          tabindex="0"
        >
          <i
            class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-popcorn"
          />
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Snacks & godteri
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          aria-label="Personlige artikler"
          class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
          data-testid="Personlige artikler"
          role="button"
          tabindex="0"
        >
          <i
            class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-soap"
          />
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Personlige artikler
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          aria-label="Kjøtt"
          class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
          data-testid="Kjøtt"
          role="button"
          tabindex="0"
        >
          <i
            class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-meat"
          />
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Kjøtt
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          aria-label="Middag"
          class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
          data-testid="Middag"
          role="button"
          tabindex="0"
        >
          <i
            class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-restaurant"
          />
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Middag
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          aria-label="Ost"
          class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
          data-testid="Ost"
          role="button"
          tabindex="0"
        >
          <i
            class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-cheese"
          />
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Ost
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          aria-label="Dessert"
          class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
          data-testid="Dessert"
          role="button"
          tabindex="0"
        >
          <i
            class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rs-pie"
          />
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Dessert
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          aria-label="Pålegg & frokost"
          class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
          data-testid="Pålegg & frokost"
          role="button"
          tabindex="0"
        >
          <i
            class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-bread-slice"
          />
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Pålegg & frokost
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          aria-label="Middagstilbehør"
          class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
          data-testid="Middagstilbehør"
          role="button"
          tabindex="0"
        >
          <i
            class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-sauce"
          />
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Middagstilbehør
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          aria-label="Drikke"
          class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
          data-testid="Drikke"
          role="button"
          tabindex="0"
        >
          <i
            class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-drink-alt"
          />
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Drikke
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
      </div>
      <h3>
        Sortering
      </h3>
      <div
        class="filtercontainer"
      >
        <div
          class="filter"
        >
          <div
            aria-label="Stigende pris"
            class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
            data-testid="Stigende pris"
            role="button"
            tabindex="0"
          >
            <i
              class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-arrow-trend-up"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Stigende pris
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            aria-label="Synkende pris"
            class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1r3mz6a-MuiButtonBase-root-MuiChip-root"
            data-testid="Synkende pris"
            role="button"
            tabindex="0"
          >
            <i
              class="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault fi fi-rr-arrow-trend-down"
            />
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Synkende pris
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="sliderContainer"
        >
          <p>
             Minimal pris
          </p>
          <div
            class="slider"
          >
            <span
              class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-187mznn-MuiSlider-root"
            >
              <span
                class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
              />
              <span
                class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
                style="left: 0%; width: 100%;"
              />
              <span
                class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
                data-index="0"
                style="left: 0%;"
              >
                <input
                  aria-orientation="horizontal"
                  aria-valuemax="200"
                  aria-valuemin="0"
                  aria-valuenow="0"
                  aria-valuetext="0"
                  data-index="0"
                  max="200"
                  min="0"
                  step="1"
                  style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                  type="range"
                  value="0"
                />
                <span
                  aria-hidden="true"
                  class="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
                >
                  <span
                    class="MuiSlider-valueLabelCircle"
                  >
                    <span
                      class="MuiSlider-valueLabelLabel"
                    >
                      0
                    </span>
                  </span>
                </span>
              </span>
              <span
                class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
                data-index="1"
                style="left: 100%;"
              >
                <input
                  aria-orientation="horizontal"
                  aria-valuemax="200"
                  aria-valuemin="0"
                  aria-valuenow="200"
                  aria-valuetext="200"
                  data-index="1"
                  max="200"
                  min="0"
                  step="1"
                  style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                  type="range"
                  value="200"
                />
                <span
                  aria-hidden="true"
                  class="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
                >
                  <span
                    class="MuiSlider-valueLabelCircle"
                  >
                    <span
                      class="MuiSlider-valueLabelLabel"
                    >
                      200
                    </span>
                  </span>
                </span>
              </span>
            </span>
          </div>
          <p>
            Maksimal pris
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
